### PR TITLE
Fix overlapping entities

### LINKED ISSUE #1562: Fix overlapping entities in evaluate script
Right now the evaluate script fails whenever there's any overlapping entity labels. It should only fail for overlapping entities *within* an entity classifier. In fact, I think it only actually makes sense for the `ner_crf`, given `ner_duckling_http` could identify both a number and a date in one which is fine.
PS. this causes evaluation on the platform to fail

### PR DESCRIPTION
**Pull request by EPedrotti**
**Proposed changes**:
- fixes #1562

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by EPedrotti** [comment]
@tmbo codeclimate is complaining about a line being more than 79 characters. Shouldn't it be set to 80?

**Comment by tmbo** [comment]
mhm good point, not sure - we do not have a codeclimate configuration so we rely on the defaults here atm.

**Comment by EPedrotti** [comment]
@tmbo can we merge this one even if codeclimate complains about the line lenght being 80 chars?

File: rasa_nlu/evaluate.py [hunk 1]
```diff
@@ -462,18 +462,19 @@ def pick_best_entity_fit(token, candidates):
         return candidates[best_fit]["entity"]
 
 
-def determine_token_labels(token, entities):
+def determine_token_labels(token, entities, extractors):
     """Determines the token label given entities that do not overlap.
 
     :param token: a single token
     :param entities: entities found by a single extractor
+    :param extractors: list of extractors
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: please convert this to use the google docstyle

```
    """Determines the token label given entities that do not overlap.

    Args:
        token: a single token
        entities: entities found by a single extractor
        extractors: list of extractors
    Returns: 
        entity type
    """
```

File: src/pip-delete-this-directory.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This file is placed here by pip to indicate the source was put
```
Comment (tmbo) [inline src/pip-delete-this-directory.txt]: please remove the files under `src/` from source control

File: rasa_nlu/evaluate.py [hunk 3]
```diff
@@ -462,18 +462,19 @@ def pick_best_entity_fit(token, candidates):
...
     :return: entity type
     """
 
     if len(entities) == 0:
         return "O"
-
-    if do_entities_overlap(entities):
+    if extractors is not None and "ner_crf" in extractors and \
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: please put the condition into a separate function, e.g.

```python
def do_extractors_support_overlap(extractors):
    """...""""
    return extractors is not None and CRFEntityExtractor.name in extractors
```

**Comment by tmbo** [comment]
Please let me know when this ready for another review

**Comment by tmbo** [comment]
@EPedrotti PLEASE request a new review when this is done, otherwise please state what needs to be done to get this to another review

**Comment by EPedrotti** [comment]
@tmbo I was on openshift and I didn't had to to review and to add tests

File: rasa_nlu/evaluate.py [hunk 4]
```diff
@@ -462,24 +463,33 @@ def pick_best_entity_fit(token, candidates):
...
         raise ValueError("The possible entities should not overlap")
 
     candidates = find_intersecting_entites(token, entities)
     return pick_best_entity_fit(token, candidates)
 
 
+def do_extractors_support_overlap(extractors):
+    """Checks if extractors support overlapping entities
+    """
+    return extractors is None or not (
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: should be `extractors is None or CRFEntityExtractor.name not in extractors`

**Pull request merged**